nvim/types.go

```diff
@@ -0,0 +1,245 @@
+// Copyright 2019 Gary Burd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nvim
+
+// Mode represents a Nvim's current mode.
+type Mode struct {
+	// Mode is the current mode.
+	Mode string `msgpack:"mode"`
+
+	// Blocking is true if Nvim is waiting for input.
+	Blocking bool `msgpack:"blocking"`
+}
+
+// HLAttrs represents a highlight definitions.
+type HLAttrs struct {
+	Bold       bool `msgpack:"bold,omitempty"`
+	Underline  bool `msgpack:"underline,omitempty"`
+	Undercurl  bool `msgpack:"undercurl,omitempty"`
+	Italic     bool `msgpack:"italic,omitempty"`
+	Reverse    bool `msgpack:"reverse,omitempty"`
+	Foreground int  `msgpack:"foreground,omitempty" empty:"-1"`
+	Background int  `msgpack:"background,omitempty" empty:"-1"`
+	Special    int  `msgpack:"special,omitempty" empty:"-1"`
+}
+
+// Mapping represents a nvim mapping options.
```

Comment:
That comment doesn't really add any information. But at least "a nvim" should be just "Nvim" (no "a").

Reply:
@justinmk I see will fix.